### PR TITLE
Update Getting Started to use Expo CLI

### PR DESCRIPTION
**Pull request by fson**
This PR implements the proposal "Use Expo CLI in the Quick Start section of the docs":
https://github.com/react-native-community/discussions-and-proposals/issues/23

* Updates Getting Started with steps to get started using Expo CLI and up-to-date instructions for getting support.
* Updates CRNA references in the docs so that they also make sense for projects created with Expo CLI.

### CODE REVIEW

**Comment by brentvatne** [comment]
lgtm!

**Comment by morenoh149** [comment]
fixes https://github.com/react-community/create-react-native-app/issues/770#issuecomment-425697225

**Comment by hramos** [comment]
Thanks for the PR! Can you also update `website/versioned_docs/version-0.5/getting-started.md` with the same contents as `docs/getting-started.md`? This will make it so that the Getting Started guide displayed remains the same regardless of which version of the docs are being browsed. You'd just need to maintain the original frontmatter YAML (with the `version-0.5-getting-started` id).

**Comment by fson** [comment]
Thanks @hramos! Updated `version-0.5` in #590. The instructions on the "Building Projects with Native Code" tab (for example, installing Android development environment) in that file had some changes that were not from this PR, so I only picked my changes.

File: docs/getting-started.md [hunk 1]
```diff
@@ -129,31 +129,33 @@ Congratulations! You've successfully run and modified your first React Native ap
...
+If you're integrating React Native into an existing project, you'll want to skip Expo CLI and go directly to setting up the native build environment. Select "Building Projects with Native Code" above for instructions on configuring a native build environment for React Native.
 
 <block class="native mac windows linux ios android" />
 
-<p>Follow these instructions if you need to build native code in your project. For example, if you are integrating React Native into an existing application, or if you "ejected" from <a href="getting-started.html" onclick="displayTab('guide', 'quickstart')">Create React Native App</a>, you'll need this section.</p>
+<p>Follow these instructions if you need to build native code in your project. For example, if you are integrating React Native into an existing application, or if you "ejected" from <a href="getting-started.html" onclick="displayTab('guide', 'quickstart')">Expo/a> or Create React Native App, you'll need this section.</p>
```
Comment (jigarakatidus) [inline docs/getting-started.md]: Typo here, it should be </a>. Doc is broken currently. Whole Paragraph is hyperlinked.

**Pull request merged**